feat: add polls

618B opened the pull request.

Commit:
feat: add polls

pingvin1308 commented on CleannetCodeBot.Twitch/Polls/PollsService.cs:
```diff
@@ -0,0 +1,112 @@
+        string rewardId;
+        try
+        {
+            var requestResult = await _twitchApi.Helix.ChannelPoints.CreateCustomRewardsAsync(broadCasterId, requestPayload, authToken);
```
Вот тут падает exception, если награда с таким названием уже существует. Думаю стоит в начале метода попробовать получить список наград и если среди них есть награда "Вопрос", то ее нужно удалить.

Также награда должна создаваться с IsEnabled = true

Commit:
fix: start testing

multiple testing questions now allowed
existing question rewards clears before creation
add authtoken check before question creation

Commit:
refactor: change schedule service to periodic timer

add logging to schedule service

Commit:
fix: remove reward before poll creation

time gap between polls does not apply if poll creation try failed by removing reward with same name

Commit:
feat: move poll settings to config

Commit:
feat: add poll result saving to mongodb

Commit:
feat: add users info saving

pingvin1308 commented on CleannetCodeBot.Twitch/CleannetCodeBot.Twitch.csproj:
```diff
@@ -26,4 +27,10 @@
         </Content>
     </ItemGroup>
 
+    <ItemGroup>
```
эту секцию нужно убрать

pingvin1308 commented on CleannetCodeBot.Twitch/Polls/Poll.cs:
```diff
@@ -0,0 +1,46 @@
+    
+    public PollQuestion Question { get; }
+    
+    public string CreatedByUser { get; }
```
CreatedByUser -> CreatedByUserId

pingvin1308 commented on CleannetCodeBot.Twitch/Polls/PollsService.cs:
```diff
@@ -0,0 +1,150 @@
+                await _twitchApi.Helix.ChannelPoints.DeleteCustomRewardAsync(broadCasterId, rewardExists.Id, authToken);
+                _logger.LogInformation($"Reward with title {title} was deleted");
+            }
+            else
```
можно убрать данный else, так как отсутствие награды нас не интересует

pingvin1308 commented on CleannetCodeBot.Twitch/TwitchWebsocketBackgroundService.cs:
```diff
@@ -178,9 +190,56 @@ private void OnChannelFollow(object? sender, ChannelFollowArgs e)
+                _logger.LogInformation($"Poll creating error: {exception.Message}");
+            }
+        }
+        else
```
Добавить проверку что награда содержит в названии "Вопрос от "

Commit:
fix: small issues
